nomenclature/codelist.py

```diff
@@ -543,8 +543,21 @@ def from_directory(cls, name: str, path: Path, file_glob_pattern: str = "**/*"):
         return cls(name=name, mapping=mapping)
 
     @property
-    def hierarchy(self):
-        raise NotImplementedError("This method is not yet implemented.")
+    def hierarchy(self) -> List[str]:
+        """Return List of strings of available hierarchies to filter
+
+        Returns
+        -------
+        List[str]
+
+        """
+
+        # creates a sorted list of strings from a set with
+        # all available hierarchy options
+        avail_ops: List[str] = sorted(
+            list({v.hierarchy for v in self.mapping.values()})
+        )
+        return avail_ops
```

Comment:
Make it one line... Also, list comprehension is easier to read, I think...
```suggestion
        return sorted([v.hierarchy for v in self.mapping.values()])
```